### PR TITLE
e2e:chore - tests for enable owasp dependency check flag

### PR DESCRIPTION
**Pull request by oliveirafelipezup**
**- What I did**
Create test for enable owasp dependecy check flag
**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: e2e/commands/start/start_test.go [hunk 1]
```diff
@@ -329,4 +328,34 @@ var _ = Describe("running binary Horusec with start parameter", func() {
...
+			flags = map[string]string{
+				testutil.StartFlagProjectPath:     projectPath,
+				testutil.StartFlagCustomRulesPath: customRulesJson,
+			}
+		})
+
+		It("Checks if the custom rules path property was set", func() {
+			Expect(session.Out.Contents()).To(ContainSubstring(fmt.Sprintf(`\"custom_rules_path\": \"%s\"`, testutil.NormalizePathToAssertInJSON(customRulesJson))))
+		})
+	})
```
Comment (matheusalcantarazup) [inline e2e/commands/start/start_test.go:345]: This test seems already been implemented on #793. Please remove from this pr

**Review by matheusalcantarazup — CHANGES_REQUESTED** [review]

File: e2e/commands/start/start_test.go [hunk 2]
```diff
@@ -380,4 +379,19 @@ var _ = Describe("running binary Horusec with start parameter", func() {
...
+
+		BeforeEach(func() {
+			flags = map[string]string{
+				testutil.StartFlagProjectPath:                projectPath,
+				testutil.StartFlagEnableOwaspDependencyCheck: oaspDependencyCheck,
+			}
+		})
+
+		It("Checks if the enable owasp dependecy was set with true", func() {
+			Expect(session.Out.Contents()).To(ContainSubstring(fmt.Sprintf(`\"enable_owasp_dependency_check\": %s,\n`, testutil.NormalizePathToAssertInJSON(oaspDependencyCheck))))
```
Comment (lucasbrunozup) [inline e2e/commands/start/start_test.go:394]: The `testutil.NormalizePathToAssertInJSON` function is unnecessary!

**Review by lucasbrunozup — CHANGES_REQUESTED** [review]

File: e2e/commands/start/start_test.go [hunk 3]
```diff
@@ -394,17 +394,18 @@ var _ = Describe("running binary Horusec with start parameter", func() {
 		})
 	})
 
-	When("--information-severity is passed", func() {
+	When("--enable-owasp-dependency-check is passed", func() {
+		oaspDependencyCheck := "true"
```
Comment (nathanmartinszup) [inline e2e/commands/start/start_test.go:398]: Typo in `oaspDependencyCheck` -> `owaspDependencyCheck`

File: e2e/commands/start/start_test.go [hunk 4]
```diff
@@ -394,17 +394,18 @@ var _ = Describe("running binary Horusec with start parameter", func() {
...
+
 		BeforeEach(func() {
 			flags = map[string]string{
-				testutil.StartFlagProjectPath:         testutil.JavaScriptExample1,
-				testutil.StartFlagInformationSeverity: "true",
+				testutil.StartFlagProjectPath:                projectPath,
+				testutil.StartFlagEnableOwaspDependencyCheck: oaspDependencyCheck,
```
Comment (nathanmartinszup) [inline e2e/commands/start/start_test.go:403]: Here too `oaspDependencyCheck` -> `owaspDependencyCheck`

**Review by wiliansilvazup — APPROVED** [review]

**Review by matheusalcantarazup — APPROVED** [review]

File: e2e/commands/start/start_test.go [hunk 5]
```diff
@@ -422,17 +422,18 @@ var _ = Describe("running binary Horusec with start parameter", func() {
 		})
 	})
 
-	When("--information-severity is passed", func() {
```
Comment (lucasbrunozup) [inline e2e/commands/start/start_test.go:425]: Why was this test deleted?

**Review by lucasbrunozup — CHANGES_REQUESTED** [review]

File: e2e/commands/start/start_test.go (hunk 5 above)
Comment (wiliansilvazup) [reply e2e/commands/start/start_test.go:425]: @oliveirafelipezup

File: e2e/commands/start/start_test.go [hunk 6]
```diff
@@ -434,6 +434,21 @@ var _ = Describe("running binary Horusec with start parameter", func() {
...
+
+		BeforeEach(func() {
+			flags = map[string]string{
+				testutil.StartFlagProjectPath:                projectPath,
+				testutil.StartFlagEnableOwaspDependencyCheck: owaspDependencyCheck,
+			}
+		})
+
+		It("Checks if the enable owasp dependecy was set with true", func() {
+			Expect(session.Out.Contents()).To(ContainSubstring(fmt.Sprintf(`\"enable_owasp_dependency_check\": %s,\n`, owaspDependencyCheck)))
```
Comment (matheusalcantarazup) [inline e2e/commands/start/start_test.go:448]: Can we add a new Expected that check that the log message "Running OwaspDependencyCheck" is printed? With this we can sure that dependency check is started.

Comment (matheusalcantarazup) [inline e2e/commands/start/start_test.go:448]: Can we add a new Expected that check that the log message "Running OwaspDependencyCheck" is printed? With this we can sure that dependency check is started.

**Review by matheusalcantarazup — CHANGES_REQUESTED** [review]

File: e2e/commands/start/start_test.go (hunk 6 above)
Comment (matheusalcantarazup) [reply e2e/commands/start/start_test.go:448]: The assert added is confusing. If the test is about checking that dependency check is running why `Expect(session.Out.Contents()).To(ContainSubstring("{HORUSEC_CLI} The tool was ignored for run in this analysis: OwaspDependencyCheck"))` was added instead `Expect(session.Out.Contents()).To(ContainSubstring("Running OwaspDependencyCheck"))`?

**Review by lucasbrunozup — APPROVED** [review]

**Comment by wiliansilvazup** [comment]
I'm going to close this PR for now, because as it's saying in [this code](https://github.com/ZupIT/horusec/blob/79234a0ea90a055349fde1dbc423084340c24a9a/internal/services/formatters/generic/dependency_check/formatter.go#L44), the owasp dependency check tool will only run if enable Docker in the analysis. So this implementation is not valid for e2e tests :)

**Pull request closed without merging**